src/Classes/ServiceAPI/MyRadio_User.php

```diff
@@ -2052,6 +2078,93 @@ public function grantPermission($authid, $from = null, $to = null)
         }
     }
 
+    /**
+     * Deactivates this user (EOL Tier 1). They will not be able to sign in, receive email, receive credits for
+     * any shows, and any officerships will be ended.
+     *
+     * Note: DO NOT CALL THIS DIRECTLY outside of the relevant daemon! Instead use requestDeactivation, as it
+     * properly enqueues the request and sends a warning email.
+     */
+    public function deactivate()
+    {
+        // Start a transaction.
+        self::$db->query('BEGIN');
+
+        // Update profile
+        self::$db->query('
+            UPDATE public.member
+            SET eol_state = $2,
+                receive_email = FALSE,
+                account_locked = TRUE
+            WHERE memberid = $1
+            ', [
+            $this->getID(),
+            MyRadio_User::EOL_STATE_DEACTIVATED
+        ]);
+
+        // End all officerships
+        self::$db->query('
+            UPDATE public.member_officer
+            SET till_date = NOW()
+            WHERE memberid = $1
+            AND till_date IS NULL
+            ', [
+            $this->getID()
+        ]);
+
+        // Unsub from all email
+        self::$db->query('
+            DELETE FROM public.mail_subscription
+            WHERE memberid = $1
+            AND (SELECT subscribable FROM public.mail_list WHERE mail_list.listid = mail_subscription.listid)
+            ', [
+            $this->getID()
+        ]);
+
+        // Don't forget!
+        self::$db->query('COMMIT');
+
+        // Sync up local changes
+        $this->eol_state = MyRadio_User::EOL_STATE_DEACTIVATED;
+        $this->receive_email = false;
+        $this->account_locked = true;
+
+        // Yeet the entire cache
+        self::$cache->purge();
+
+        // And send a goodbye email
+        // We use mail() manually to avoid hacking with the innards of MyRadioEmail
+        // (as it'd refuse to send due to receive_email being false)
+        // TODO: is this a good idea?
+        $domain = Config::$email_domain;
+        $long_name = Config::$long_name;
+        $short_name = Config::$short_name;
+        mail(
```

Comment:
There must be an existing send mail function somewhere

Reply:
The comment immediately above:

> We use mail() manually to avoid hacking with the innards of MyRadioEmail (as it'd refuse to send due to receive_email being false)

You do have a valid point though, and it may be better to refactor MRE to allow "send immediately, even if they've opted out".